pkg/skaffold/lint/k8smanifests.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lint
+
+import (
+	"context"
+	"io/ioutil"
+	"path/filepath"
+
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/config"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/output/log"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+// for testing
+var k8sManifestRules = &k8sManifestLintRules
+
+var K8sManifestLinters = []Linter{
+	&RegExpLinter{},
+	&YamlFieldLinter{},
+}
+
+var k8sManifestLintRules = []Rule{
+	{
+		Filter: YamlFieldFilter{
+			Filter: yaml.Lookup("metadata", "labels", "app.kubernetes.io/managed-by"),
+		},
+		RuleID:   K8sManifestManagedByLabelInUse,
+		RuleType: YamlFieldLintRule,
+		ExplanationTemplate: "Found usage of label 'app.kubernetes.io/managed-by'.  skaffold overwrites the 'app.kubernetes.io/managed-by' field to 'app.kubernetes.io/managed-by: skaffold'. " +
+			"and as such is recommended to remove this label",
+	},
+}
+
+func GetK8sManifestsLintResults(ctx context.Context, opts Options) (*[]Result, error) {
+	cfgs, err := getConfigSet(ctx, config.SkaffoldOptions{
+		ConfigurationFile:   opts.Filename,
+		ConfigurationFilter: opts.Modules,
+		RepoCacheDir:        opts.RepoCacheDir,
+		Profiles:            opts.Profiles,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	l := []Result{}
+	workdir, err := realWorkDir()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, c := range cfgs {
+		for _, pattern := range c.Deploy.KubectlDeploy.Manifests {
+			// NOTE: pattern is a pattern that can have wildcards, eg: leeroy-app/kubernetes/*
+			if util.IsURL(pattern) {
+				log.Entry(ctx).Infof("skaffold lint found url manifest and is skipping lint rules for: %s", pattern)
```

Comment:
shd we move this to debug instead? Additional log lines to users for default level.

Reply:
good point, done